pkg/types/installconfig.go

```diff
@@ -7,6 +7,15 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// PlatformNameAWS is name for AWS platform.
+	PlatformNameAWS string = "aws"
```

Comment:
If we're going back to enumerating these (for compile-time type checking?), then I think we might as well restore the `Platform` type we dropped [here][1] (although we'd have to call it something else now that we have a `Platform` structure.  Maybe `PlatformType`?).

[1]: https://github.com/openshift/installer/commit/619db9259e7b209795858031a895cc73e84b540e#diff-624824fc4f591fd072f8089faefc8ce6L28

Reply:
> If we're going back to enumerating these (for compile-time type checking?)
just constants so that i'm not writing "aws" everywhere.

https://github.com/openshift/installer/commit/619db9259e7b209795858031a895cc73e84b540e#diff-624824fc4f591fd072f8089faefc8ce6L28 was useful when we asked users for name, this is not required now with `*fields`.

Reply:
IMO:
The compile time validation etc. are only useful if these are inputs to installer functions; But these constants are only meant for utility for consumers who don't want to think is returned is `aws`/`AWS` `openstack`/`Openstack` etc..

Yeah, if we every make them inputs to functions, we should add validations then.